clients/pkg/logentry/stages/drop.go

```diff
@@ -159,15 +159,22 @@ func (m *dropStage) shouldDrop(e Entry) bool {
 					level.Debug(m.logger).Log("msg", "line met drop criteria for finding source key in extracted map")
 				}
 			} else {
-				if *m.cfg.Value == v {
+				s, err := getString(v)
```

Comment:
Although I think this is an improvement, it means we won't drop lines in scenarios were the parsing didn't work, which is different from the current scenario. Because of it, I think this should be documented in our CHANGELOG.